js/releases.js

```diff
@@ -0,0 +1,126 @@
+import { Octokit } from "https://cdn.skypack.dev/@octokit/rest";
+const octokit = new Octokit({});
+
+function releaseName(release) {
+    const result = document.createElement("td");
+    result.innerText = release.name;
+
+    if (release.prerelease) {
+        const tag = document.createElement("code");
+        result.innerText += " ";
+        tag.innerText = "preview";
+        tag.classList.add("ml-1");
+        result.appendChild(tag);
+    }
+
+    return result;
+}
+
+function publishDate(release) {
+    const result = document.createElement("td");
+
+    const date = new Date(release.published_at);
+    const year = date.getFullYear();
+    const month = (date.getMonth() + 1).toString().padStart(2, "0");
+    const day = date.getDate().toString().padStart(2, "0");
+    result.innerText = year + "-" + month + "-" + day;
+
+    return result;
+}
+
+function downloadLink(release, name, filter) {
+    const asset = filter ? release.assets.find(item => item.name.indexOf(filter) > -1) : undefined;
+    const result = document.createElement("li");
+    if (asset) {
+        const link = document.createElement("a");
+        link.innerText = name;
+        link.target = "_blank";
+        link.href = asset.browser_download_url;
+        result.appendChild(link);
+    } else {
+        result.innerText = name;
+        result.style.textDecoration = "line-through";
+    }
+    return result;
+}
+
+function desktopDownloads(release) {
+    const result = document.createElement("td");
+    const list = document.createElement("ul");
+    list.classList.add("row");
+
+    list.appendChild(downloadLink(release, "Linux", "-linux"));
+    list.appendChild(downloadLink(release, "Mac OS", "-macos"));
+    list.appendChild(downloadLink(release, "Windows (32-bit)", "-windows-x86_32"));
+    list.appendChild(downloadLink(release, "Windows (64-bit)", "-windows-x86_64"));
+
+    result.appendChild(list);
+    return result;
+}
+
+function browserDownloads(release) {
+    const result = document.createElement("td");
+    const list = document.createElement("ul");
+    list.classList.add("row");
+
+    list.appendChild(downloadLink(release, "Firefox", "-firefox-unsigned"));
+    list.appendChild(downloadLink(release, "Chrome / Edge", "-extension."));
+    list.appendChild(downloadLink(release, "Safari", false));
+
+    result.appendChild(list);
+    return result;
+}
+
+function webDownloads(release) {
+    const result = document.createElement("td");
+    const list = document.createElement("ul");
+    list.classList.add("row");
+
+    list.appendChild(downloadLink(release, "Self Hosted", "-selfhosted"));
+
+    result.appendChild(list);
+    return result;
+}
+
+function updateTable(table_id, releases, limit, withDate) {
+    const table = document.getElementById(table_id);
+    const entries = document.getElementById(table_id + "_entries");
+    if (releases.length === 0) {
+        table.style.display = "none";

```

Comment:
Am I reading this wrong, or will this hide the table if the JS API fails, even if Jekyll previously succeeded?

Reply:
It won't get this far if it fails, it'll throw and escape the code

Reply:
Makes sense, I didn't realize octokit threw on failure.